ddl/ddl_db_test.go

```diff
@@ -547,28 +554,71 @@ func sessionExec(c *C, s kv.Storage, sql string) {
 	se.Close()
 }
 
+func sessionExecInGoroutine(c *C, s kv.Storage, sql string, done chan *errorParams) {
+	go func() {
+		se, err := tidb.CreateSession(s)
+		defer se.Close()
+		if err != nil {
+			done <- &errorParams{
+				err,
+				IsNil,
+				nil,
+			}
+			return
+		}
+		_, err = se.Execute("use test_db")
+		if err != nil {
+			done <- &errorParams{
+				err,
+				IsNil,
+				nil,
+			}
+			return
+		}
+		rs, err := se.Execute(sql)
+		if err != nil {
+			done <- &errorParams{
+				err,
+				IsNil,
+				[]interface{}{Commentf("err:%v", errors.ErrorStack(err))},
+			}
+			return
+		}
+		if rs != nil {
+			done <- &errorParams{
+				rs,
+				IsNil,
+				nil,
+			}
+			return
+		}
+		done <- nil
+	}()
+}
+
 func (s *testDBSuite) testAddColumn(c *C) {
-	done := make(chan struct{}, 1)
+	done := make(chan *errorParams, 1)
 
 	num := defaultBatchSize + 10
 	// add some rows
 	for i := 0; i < num; i++ {
 		s.mustExec(c, "insert into t2 values (?, ?, ?)", i, i, i)
 	}
 
-	go func() {
-		sessionExec(c, s.store, "alter table t2 add column c4 int default -1")
-		done <- struct{}{}
-	}()
+	sessionExecInGoroutine(c, s.store, "alter table t2 add column c4 int default -1", done)
 
 	ticker := time.NewTicker(s.lease / 2)
 	defer ticker.Stop()
 	step := 10
 LOOP:
 	for {
 		select {
-		case <-done:
-			break LOOP
+		case result := <-done:
+			if result == nil {
+				break LOOP
+			} else {
```

Comment:
If `result == nil` is true, it will break `LOOP`. So the `else` is useless, remove it.